velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -2747,6 +2747,118 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, arbitrateMemoryFromOtherOperator) {
   }
 }
 
+DEBUG_ONLY_TEST_F(SharedArbitrationTest, joinBuildSpillError) {
```

Comment:
What does this test do?

Should we create separate test for different operators? This test file is already 3k lines long.

Reply:
This test is to verify the reclaim error is handled properly. We could separate them into separate file one per each operator. And move SharedArbitrator and SharedArbitratorTest into separate files.

Reply:
> This test is to verify the reclaim error is handled properly.

Would you explain how it does that? It is a bit hard to read the test code.

Reply:
The test is to trigger spilling failure during hash join spill processing. I simplify it a bit.